analyticsclient/tests/__init__.py

```diff
@@ -49,67 +49,102 @@ def verify_last_querystring_equal(self, expected_query):
-            httpretty.register_uri(httpretty.GET, uri, body='{}')
-            getattr(self.client_class, self.endpoint)(**kwargs)
-            self.verify_last_querystring_equal(self.expected_query(**kwargs))
-
-    def test_all_items_url(self):
-        """Endpoint can be called without parameters."""
-        httpretty.register_uri(
-            httpretty.POST if self.uses_post_method else httpretty.GET,
-            self.base_uri, body='{}'
-        )
-        getattr(self.client_class, self.endpoint)()
+
+        uri_template = '{uri}?'
+        for key in kwargs:
+            uri_template += '%s={%s}' % (key, key)
+        uri = uri_template.format(uri=self.base_uri, **kwargs)
+
+        httpretty.register_uri(httpretty.GET, uri, body='{}')
+        getattr(self.client_class, self.endpoint)(**kwargs)
+        self.verify_last_querystring_equal(self.expected_query(**kwargs))
+
+    def fill_in_empty_params_with_dummies(self, **kwargs):
+        """Fill in non-provided parameters with dummy values so they are tested."""
+        params = {param: '.' for param in self.other_params}
+        params.update(kwargs)
+        return params
 
     @ddt.data(
+        [],
         ['edx/demo/course'],
-        ['edx/demo/course', 'another/demo/course']
+        ['edx/demo/course', 'another/demo/course'],
     )
-    def test_courses_ids(self, ids):
-        """Endpoint can be called with IDs."""
-        self.kwarg_test(**{self.id_field: ids})
+    def test_url_with_params(self, ids):
+        """Endpoint can be called with parameters, including IDs."""
+        params = self.fill_in_empty_params_with_dummies(**{self.id_field: ids})
+        self.verify_query_params(**params)
+
+    def test_url_without_params(self):
+        """Endpoint can be called without parameters."""
+        httpretty.register_uri(httpretty.GET, self.base_uri, body='{}')
+        getattr(self.client_class, self.endpoint)()
+
+
+class APIWithPostableIDsTestCase(APIWithIDsTestCase):
+    """Base class for tests for API endpoints that can POST a list of course IDs."""
+
+    @httpretty.activate
+    def verify_post_data(self, **kwargs):
+        """Construct POST request with parameters and check if it is what we expect."""
+        httpretty.reset()
+        httpretty.register_uri(httpretty.POST, self.base_uri, body='{}')
+        getattr(self.client_class, self.endpoint)(**kwargs)
+
+        expected_body = kwargs.copy()
+        for key, val in expected_body.iteritems():
+            if not isinstance(val, list):
+                expected_body[key] = [val]
+        actual_body = httpretty.last_request().parsed_body
+        self.assertDictEqual(actual_body or {}, expected_body)
+
+    def test_request_with_many_ids(self):
+        """Endpoint can be called with a large number of ID parameters."""
+        params = self.fill_in_empty_params_with_dummies(**{self.id_field: ['id'] * 10000})
+        self.verify_post_data(**params)
```

Comment:
[This](https://github.com/edx/edx-analytics-data-api-client/pull/36/files/2ffb6e64e8b7b18f748e0ddbf4625f03bda5ef21#diff-d8b9f032ba99ee92e0627861280e7fadR85) test takes care of that.